.codecov.yml

```diff
@@ -8,3 +8,6 @@ coverage:
         threshold: 0.85
         base: auto
     patch: off
+
+ignore:
+  - "[a-z]+/setup.py"  # Ignore all setup.py files
```

Comment:
I actually had a discussion about this with Tim before going ahead and ignoring setup.py files explicitly. Here are the reasons:
* `setup.py` files will contain a lot of boilerplate that is hard to test and usually is never tested. Also, with our automation code, this increases the lines of boilerplate needed to install the packages in different ways.
* We don't necessarily need to optimize *every* file for test coverage, certainly not `setup.py` files or scripts like `release_packages`.
* The build was failing because of codecov, so it didn't seem right to merge something that was explicitly failing. So this seemed like a good way to say that its intended to fail.

What do you think? :)

Reply:
I don't understand this part:
> Also, with our automation code, this increases the lines of boilerplate needed to install the packages in different ways.

Apart from that, I agree with the first two points.

For the third point, I think it's fine if codecov fails for the build, since it only tests coverage and we know that the reason is negligible in this case. Future builds won't fail, since their coverage change will be calculated relative to this commit. Furthermore we aren't near a coverage percentage where it would make sense to exclude files for the sake of keeping the number high. 

All in all, I don't really see a reason to exclude it. I also don't see any reason not to exclude it, other than three additional lines. Thus, I'd vote for keeping codecov as it is, but it's not a strong opinion.

Reply:
Cool, I'm convinced, we won't exclude it then! :)

What I meant by:
``` quote
Also, with our automation code, this increases the lines of boilerplate needed to install the packages in different ways.
```
is that with the release script, I had to add a bunch of boilerplate to `setup.py` files that triggered the codecov warnings, because that code wasn't tested.